### PR TITLE
CORTX-30131: Automated RADOS CI/CD - PART1 (IO Sanity & Build Versioning).

### PR DESCRIPTION
**Pull request by nitisdev**
Signed-off-by: Nitish Singh <nitish.singh@seagate.com>

# Problem Statement
- Automated IO Sanity and versioning.

# Design
-  IO Sanity
- Build Vesioning

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- End to end deployment:
- ceph-build: (container)
rockylinux: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/ceph-build-container/82
ubuntu: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/ceph-build-container/85
centos: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/ceph-build-container/83
- Build Versioning Artifacts: http://cortx-storage.colo.seagate.com/releases/ceph/
- Ceph deploy: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/ceph-deploy/46/console
- setup-cortx-rgw-cluster: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/5483/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by gauravchaudhari02 — APPROVED** [review]

File: solutions/kubernetes/ceph-binary-build.sh [hunk 1]
```diff
@@ -19,8 +19,10 @@
 #
 
 source functions.sh
+source /etc/os-release
 
 BUILD_LOCATION="$2"
+mount="cortx-storage.colo.seagate.com:/mnt/data1/releases/ceph"
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:25]: Can we make this as input parameter

File: solutions/kubernetes/ceph-deploy.sh [hunk 2]
```diff
@@ -25,6 +25,7 @@ OSD_DISKS=$PWD/osd_disks
 SSH_KEY_FILE=/root/.ssh/id_rsa
 ALL_NODES=$(cat "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
 PRIMARY_NODE=$(head -1 "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
+CEPH_DEPLOYMENT="True"
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-deploy.sh:28]: Please change `True` to `true`

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: solutions/kubernetes/ceph-binary-build.sh (hunk 1 above)
Comment (nitisdev) [reply solutions/kubernetes/ceph-binary-build.sh]: Done

File: solutions/kubernetes/ceph-deploy.sh (hunk 2 above)
Comment (nitisdev) [reply solutions/kubernetes/ceph-deploy.sh]: Done

**Review by shailesh-vaidya — DISMISSED** [review]

File: jenkins/automation/rgw/ceph/ceph-build-docker.groovy [hunk 3]
```diff
@@ -13,14 +13,19 @@ pipeline {
...
+
     parameters {
         string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for Cluster Setup scripts', trim: true)
         string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
         string(name: 'CEPH_REPO', defaultValue: 'https://github.com/ceph/ceph/', description: 'Repository for Cluster Setup scripts', trim: true)
         string(name: 'CEPH_BRANCH', defaultValue: 'quincy', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
         choice(
             name: 'BUILD_OS',
-            choices: ['Ubuntu', 'CentOS', 'Rocky Linux'],
+            choices: ['ubuntu-20.04', 'centos-8', 'rockylinux-8.4'],
```
Comment (gauravchaudhari02) [inline jenkins/automation/rgw/ceph/ceph-build-docker.groovy:28]: lets keep `rockylinux-8.4` as default

File: jenkins/automation/rgw/ceph/ceph-build-vm.groovy [hunk 4]
```diff
@@ -13,14 +13,19 @@ pipeline {
...
+
     parameters {
         string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for Cluster Setup scripts', trim: true)
         string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
         string(name: 'CEPH_REPO', defaultValue: 'https://github.com/ceph/ceph/', description: 'Repository for Cluster Setup scripts', trim: true)
         string(name: 'CEPH_BRANCH', defaultValue: 'quincy', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
         choice(
             name: 'BUILD_OS',
-            choices: ['Ubuntu', 'CentOS', 'Rocky Linux'],
+            choices: ['ubuntu-20.04', 'centos-8', 'rockylinux-8.4'],
```
Comment (gauravchaudhari02) [inline jenkins/automation/rgw/ceph/ceph-build-vm.groovy:28]: lets keep `rockylinux-8.4` as default

Comment (nitisdev) [reply jenkins/automation/rgw/ceph/ceph-build-vm.groovy]: Done

File: jenkins/automation/rgw/ceph/ceph-build-docker.groovy (hunk 3 above)
Comment (nitisdev) [reply jenkins/automation/rgw/ceph/ceph-build-docker.groovy]: Done

**Review by gauravchaudhari02 — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**